flax/training/common_utils.py

```diff
@@ -21,33 +25,80 @@
 
 
 def shard(xs):
+  """Helper for pmap to shard a pytree of arrays by local_device_count.
+
+  Args:
+    xs: a pytree of arrays.
+  Returns:
+    A matching pytree with arrays' leading dimensions sharded by the
+    local device count.
+  """
   local_device_count = jax.local_device_count()
   return jax.tree_util.tree_map(
       lambda x: x.reshape((local_device_count, -1) + x.shape[1:]), xs)
 
 
 def shard_prng_key(prng_key):
-  # PRNG keys can used at train time to drive stochastic modules
-  # e.g. DropOut. We would like a different PRNG key for each local
-  # device so that we end up with different random numbers on each one,
-  # hence we split our PRNG key and put the resulting keys into the batch
-  return jax.random.split(prng_key, num=jax.local_device_count())
+  """Helper to shard (aka split) a PRNGKey for use with pmap'd functions.
 
+   PRNG keys can used at train time to drive stochastic modules
```

Comment:
These lines have an additional 1 space of indentation so they are interpreted similar to markdowns `>`:
![Screenshot from 2022-08-04 19-28-40](https://user-images.githubusercontent.com/5862228/182977305-812837a4-0bc7-4a7c-9f04-9d696b2b6cc4.png)

Reply:
oops, thanks